core/src/main/java/org/apache/brooklyn/core/catalog/internal/CatalogBomScanner.java

```diff
@@ -168,11 +227,18 @@ private String addLibraryDetails(Bundle bundle, String bomText) {
                     LOG.warn("Unexpected syntax for {} (expected Map), ignoring", BROOKLYN_CATALOG);
                 }
             }
-            final String updatedBom = new Yaml().dump(bom);
+            final String updatedBom = backToYaml(bom);
```

Comment:
That is true, but the formatting added here will hopefully be good enough looking to be acceptable.  Certainly worth keeping in mind.